sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -214,11 +213,12 @@ case class Stack(children: Seq[Expression]) extends Generator {
 
     // Create the collection.
     val wrapperClass = classOf[mutable.WrappedArray[_]].getName
-    ctx.addMutableState(
-      s"$wrapperClass<InternalRow>",
-      ev.value,
-      v => s"$v = $wrapperClass$$.MODULE$$.make($rowData);", useFreshName = false)
-    ev.copy(code = code, isNull = "false")
+    ev.copy(code =
+      s"""
+         |InternalRow[] $rowData = new InternalRow[$numRows];
```

Comment:
this creates a large array every time, and I don't think we have data copy issues for generator expressions...

Reply:
I see. I did not imagine that `numRows` is large.
I will revert the code for `rowData`.